src/printers/flow.js

```diff
@@ -19,11 +19,67 @@ export default {
     ? Tokens.string('true')
     : Tokens.string('false'),
 
+  ExistentialTypeParam: () => [
+    Tokens.string('*'),
+  ],
+
+  FunctionTypeAnnotation: ({node, print, path}) => [
+    print(node.typeParameters),
+    Tokens.string('('),
+    (node.params.length || node.rest) && [
+      Tokens.scopeOpen('object'),
+      Tokens.scopeEmptyOrBreak(),
+      map(node.params, (param, i, arr) => [
+        i > 0 && Tokens.scopeSpaceOrBreak(),
+        print(param),
+        arr.length - 1 !== i && !node.rest
+          ? Tokens.comma()
+          : Tokens.scopeEmptyOrComma(),
+      ]),
+      node.rest && [
+        node.params.length && [
+          Tokens.comma(),
+          Tokens.scopeSpaceOrBreak(),
+        ],
+        Tokens.string('...'),
+        print(node.rest),
+        Tokens.scopeEmptyOrComma(),
+      ],
+      Tokens.scopeEmptyOrBreak(),
+      Tokens.scopeClose(),
+    ],
+    Tokens.string(')'),
+    (
+      path[path.length - 2].type === 'ObjectTypeProperty' ||
```

Comment:
Yeah but I don't want to encourage the use of parent, ideally all printing is based on the current node or the parent get information about the children and defers to some helper.

We could always add a utilityfn though